integration/spark/src/main/scala/org/apache/carbondata/spark/rdd/NewCarbonDataLoadRDD.scala

```diff
@@ -160,6 +161,7 @@ class NewCarbonDataLoadRDD[K, V](
         executor.execute(model,
           loader.storeLocation,
           recordReaders)
+        executor.close()
```

Comment:
good catch.

But better to add it inside taskCompletion listener. refer UpdateDataLoad.scala line 70

Reply:
it already added, but we need to invoke it before upload metrics.